src/platforms/common/stm32/serialno.c

```diff
@@ -70,3 +70,40 @@ void read_serial_number(void)
 #endif
 	serial_no[DFU_SERIAL_LENGTH - 1] = '\0';
 }
+
+char *__utoa_upper(unsigned value, char *str, int base)
```

Comment:
Given this is code outside the C runtime and implementation, `__` identifiers are expressly UB and forbidden - so, while we know why it's got those underscores, perhaps drop them in this first commit of the code so we don't accidentally break something later if any of these commits have to be bisected through or reverted?

Reply:
Absolutely. All commits regarding introducing this function will be squashed and rebased into one. I just wanted to preserve development history so you can judge with having better visibility.